airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_message.yaml

```diff
@@ -118,10 +118,43 @@ definitions:
       name:
         type: string
         description: Stream's name.
-      json_schema:
-        description: Stream schema using Json Schema specs.
+      discovered:
+        description: Contains information about the data in the source. This field is set by the source and should be read-only thereafter.
         type: object
-        existingJavaType: com.fasterxml.jackson.databind.JsonNode
+        properties:
+          json_schema:
+            description: Stream schema using Json Schema specs.
+            type: object
+            existingJavaType: com.fasterxml.jackson.databind.JsonNode
+          metadata:
+            description: Additional information about the data in the source, including information about what Airbyte features it supports.
+            type: object
+            properties:
+              supported_sync_modes:
+                type: array
+                items:
+                  "$ref": "#/definitions/SyncMode"
+                # backwards compatibility: if not present, assume full_refresh only.
```

Comment:
```suggestion
```
This seems to be a reasonable default even without the excuse of backwards compatibility.